rocketchat/templates/scripts-configmap.yaml

```diff
@@ -39,4 +39,110 @@ data:
       quit(1);
     }
     quit(0);
-{{- end }}
\ No newline at end of file
+  updateSynapseHomeserverConfig.sh: |
```

Comment:
This is very bitnami-ish. But, unfortunately, no way around it.

For synapse we have
1. synapse generates config in initContainer
2. we append our own config into it
3. synapse starts

If we change our own config later, we wouldn't want to just append but replace whatever we have already. 

If users need to add their config, there needs to be a way to add without forever appending across restarts or recreations.

This stores the hashes of each "extra config" alongside primary config, if hash changes, **replaces** the part of the config with the new one. That's all.

PS we definitely need to have an "extra config" feature in since postgres passwords can't be passed via evironment variables, thus if postgres secrets have to live outside of values.yaml, they need to be injected via another secret.